packages/grid/x-data-grid/src/hooks/features/columns/useGridColumnGrouping.ts

```diff
@@ -0,0 +1,186 @@
+import * as React from 'react';
+import { GridApiCommunity } from '../../../models/api/gridApiCommunity';
+import { DataGridProcessedProps } from '../../../models/props/DataGridProps';
+import { GridStateInitializer } from '../../utils/useGridInitializeState';
+import {
+  GridColumnGroupingModel,
+  GridColumnNode,
+  isLeaf,
+} from '../../../models/gridColumnGrouping';
+import { gridColumnGroupsLookupSelector, gridColumnLookupSelector } from './gridColumnsSelector';
+import { GridColumnGroupLookup } from './gridColumnsInterfaces';
+import { GridColumnGroupingApi } from '../../../models/api/gridColumnGroupingApi';
+import { useGridApiMethod } from '../../utils/useGridApiMethod';
+import { GridStateColDef, GridColDef } from '../../../models/colDef';
+
+export function hasGroupPath(
+  lookupElement: GridColDef | GridStateColDef,
+): lookupElement is GridStateColDef {
+  return (<GridStateColDef>lookupElement).groupPath !== undefined;
+}
+
+type UnwrappedGroupingModel = { [key: string]: string[] };
+
+// The is the recurrence function that help writing `unwrapGroupingColumnModel()`
+const recurrentUnwrapGroupingColumnModel = (
+  columnGroupNode: GridColumnNode,
+  parents: any,
+): UnwrappedGroupingModel => {
+  if (isLeaf(columnGroupNode)) {
```

Comment:
The `isLeaf` helper function seems to be used inside this file. Maybe we can keep it here and add it to the public API, at least for now?

Reply:
This function is so small that developpers will be faster to rewrite it than checking in the doc its name

```ts
export function isLeaf(node: GridColumnNode): node is GridLeafColumn {
  return (<GridLeafColumn>node).field !== undefined;
}
```